test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -204,10 +223,296 @@ describe('IexecPoco1', () => {
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x04',
+            );
+        });
+
+        it('[Standard] Should not match orders with bad request order workerpool price', async () => {
+            orders.requester.workerpoolmaxprice = Number(orders.workerpool.workerpoolprice) - 1;
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x05',
+            );
+        });
+
+        it('[Standard] Should not match orders with bad request order tag', async () => {
+            orders.requester.tag = toBytes32('0x1101'); // Random, bytes32 short form.
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x06',
+            );
+        });
+
+        it('[Standard] Should not match orders with bad app order tag', async () => {
+            orders.app.tag = toBytes32('0x1101'); // Random, bytes32 short form.
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x06',
+            );
+        });
+
+        it('[Standard] Should not match orders with bad dataset order tag', async () => {
+            orders.app.tag = toBytes32('0x1101'); // Random, bytes32 short form.
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x06',
+            );
+        });
+
+        it('[Standard] Should not match orders with bad workerpool order tag', async () => {
+            orders.app.tag = toBytes32('0x1000'); // Random
+            orders.dataset.tag = toBytes32('0x1100'); // Random
+            orders.requester.tag = toBytes32('0x1010'); // Random
+            // Workerpool order has to satisfy conditions of different actors.
+            orders.workerpool.tag = toBytes32('0x1100'); // Bad tag, correct tag should be 0x001110
+            // Sign and match orders.
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x06',
+            );
+        });
+
+        it('[Standard] Should not match orders when TEE tag is not requested by app', async () => {
```

Comment:
Can you please update considering `Boost#it('Should fail when the last bit of app tag does not provide what dataset or request expect'`?